src/postgres-util/src/tunnel.rs

```diff
@@ -23,6 +23,42 @@ use tracing::{info, warn};
 
 use crate::PostgresError;
 
+pub async fn available_replication_slots(
+    ssh_tunnel_manager: &SshTunnelManager,
+    config: &Config,
+) -> Result<i64, PostgresError> {
+    let client = config
+        .connect("postgres_check_replication_slots", ssh_tunnel_manager)
+        .await?;
+
+    let available_replication_slots = client
+        .query_one(
+            "SELECT
+            CAST(max_replication_slots AS int8) - current_replication_slots
+            AS available_replication_slots
+        FROM
+            (
+            SELECT
+                count(*) AS current_replication_slots,
+                (
+                    SELECT setting
+                    FROM pg_settings
+                    WHERE name = 'max_replication_slots'
+                )
+                AS max_replication_slots
+            FROM pg_catalog.pg_replication_slots
+            )
+            AS o;",
+            &[],
```

Comment:
I think there's a simpler way to write this query. Maybe something like this?

```sql
SELECT current_setting('max_replication_slots')::int8 - (SELECT count(*) FROM pg_catalog.pg_replication_slots)
```

Reply:
Ah, I'd never come across `current_setting`--that was what I was looking for. ty.

Reply:
🫡